onnxruntime/contrib_ops/cuda/decoder/fastertransformer_decoder_attention/decoder_masked_multihead_attention_impl.cu

```diff
@@ -304,7 +304,7 @@ __global__ void masked_multihead_attention_kernel(DecoderMaskedMultiHeadAttentio
   int ti_end = ((tlength + K_PER_WARP - 1) / K_PER_WARP) * K_PER_WARP;
 
   // Iterate over the keys/timesteps to compute the various (Q*K^T)_{ti} values.
-  bool has_beams = params.cache_indir != nullptr;
+  bool has_beams = params.cache_indir != nullptr && !params.is_cross_attention;
```

Comment:
Don't we support beam search with cross attention ? If so, how does it work with encoder-decoder architectures like T5 which by definiton have cross attention ?

Reply:
The cross attention takes in static key/value cache and they are not involved in beam-indices based buffer switch.